src/dialect/onnx/onnx_ops.cpp

```diff
@@ -656,6 +656,11 @@ void ONNXReshapeOp::inferShapes() {
   if (outputRank < 0)
     emitError("Shape tensor must have constant shape");
 
+  // Compute total number of elements.
+  int64_t totalInputSize = 1;
+  for(auto inputDim : inputTensorTy.getShape())
+    totalInputSize *= inputDim;
+
```

Comment:
looking at the definition of secondArgDefiningOp... if I look at the auto generated code, I see this:

```
Value  ONNXReshapeOpOperandAdaptor::data() {
  return *getODSOperands(0).begin();
}
```

couldn't we use that below? It would be much more portable and readable.

I would also suggest to add a function that would be invoked like this:
```CastOrNull< mlir::ONNXConstantOp>(data()) ```
to hide some of the ugliness of this, as I expect that we would use this quite a bit. You should then use the 
```
static size_t ArrayAttrSize(ArrayAttr a) { return a.size(); }

static size_t ArrayAttrSize(Optional<ArrayAttr> a) {
  return a.getValue().size();
}

static int64_t ArrayAttrIntVal(ArrayAttr a, int i) {
  return (a.getValue()[i]).cast<IntegerAttr>().getInt();
}

static int64_t ArrayAttrIntVal(Optional<ArrayAttr> a, int i) {
  return (a.getValue().getValue()[i]).cast<IntegerAttr>().getInt();
}
```
interfaces